Source/Initialization/DivCleaner/ProjectionDivCleaner.cpp

```diff
@@ -288,34 +359,53 @@ ProjectionDivCleaner::correctBfield ()
                 Bz_arr(i,j,0) += CylindricalYeeAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, 0, 0);
             });
 #else
-            amrex::ParallelFor(tbx, tby, tbz,
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                Bx_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDx(sol_arr, coefs_x, n_coefs_x, i, j, k);
-            },
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                By_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDy(sol_arr, coefs_y, n_coefs_y, i, j, k);
-            },
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                Bz_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, k);
-            });
+            if (m_grid_type == GridType::Staggered) {
+                amrex::ParallelFor(tbx, tby, tbz,
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    Bx_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDx(sol_arr, coefs_x, n_coefs_x, i, j, k);
+                },
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    By_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDy(sol_arr, coefs_y, n_coefs_y, i, j, k);
+                },
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    Bz_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, k);
+                });
+            } else {
+                amrex::ParallelFor(tbx, tby, tbz,
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    Bx_arr(i,j,k) += CartesianNodalAlgorithm::DownwardDx(sol_arr, coefs_x, n_coefs_x, i, j, k);
+                },
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    By_arr(i,j,k) += CartesianNodalAlgorithm::DownwardDy(sol_arr, coefs_y, n_coefs_y, i, j, k);
+                },
+                [=] AMREX_GPU_DEVICE (int i, int j, int k)
+                {
+                    Bz_arr(i,j,k) += CartesianNodalAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, k);
+                });
+            }
 #endif
         }
         // Synchronize the ghost cells, do halo exchange
         ablastr::utils::communication::FillBoundary(*Bx,
                                                     Bx->nGrowVect(),
                                                     WarpX::do_single_precision_comms,
-                                                    geom[ilev].periodicity());
+                                                    geom[ilev].periodicity(),
+                                                    true);
```

Comment:
Do I understand correctly that this `true` is just making the `FillBoundary` call blocking (to synchronize threads)? If so it would make sense to only make the `Bz` call synchronize, no?

Reply:
the true is for nodal synchronization, it is still asynchronous.